js/main.js

```diff
@@ -146,9 +147,47 @@ let checkbox_initializer = (thing, e) => {
     checkbox.addEventListener("change", cb);
 };
 
+let plusminus_initializer = (thing, e) => {
+    let div = e.querySelector("div.thing-detail");
+    let plus = e.querySelector("span[name='plus']");
+    let minus = e.querySelector("span[name='minus']");
+    let value_ele = e.querySelector("span[name='value']");
+    let new_value;
+    plus.addEventListener('click', () => {
+        //plus.click()
+        div.classList.add("pending");
+        new_value = parseInt(value_ele.innerText) + 1
+        console.log(new_value);
+        cb()
+    });
+    minus.addEventListener('click', () => {
+        //minus.click();
+        div.classList.add("pending");
+        new_value = parseInt(value_ele.innerText) - 1;
+        console.log(value_ele.innerText);
+        cb()
+    });
+
+    let cb = () => {
+
+        console.log(new_value);
+        socket.send(JSON.stringify({
+            type: "command",
+            id: thing.id,
+            value: new_value,
+        }));
+        add_pending_change(thing, setTimeout(() => {
+            value_ele.innerText = new_value;
+            div.classList.remove("pending");
+        }, 1000));
+        div.classList.remove("pending");
+    };
```

Comment:
It would be better to move the definition above the usages in the plus and minus button event handlers above. It confused me and I first thought this code was calling the `cb` defined above for a different kind of handler.